docs/content/contributing/monorepo.md

```diff
@@ -0,0 +1,103 @@
+---
+description: >
+    Monorepo structure for kcp projects.
+---
+
+# Monorepo Structure
+
+The [kcp-dev/kcp](https://github.com/kcp-dev/kcp) repository is a monorepo that contains the source code for:
+
+- the kcp core
+- libraries that are used in the kcp core or are close to the kcp core
+
+At the moment, the following libraries are part of the monorepo:
+
+- [kcp-dev/apimachinery](https://github.com/kcp-dev/apimachinery)
+- [kcp-dev/client-go](https://github.com/kcp-dev/client-go)
+- [kcp-dev/code-generator](https://github.com/kcp-dev/code-generator)
+
+## Terminology
+
+- The monorepo: the [kcp-dev/kcp](https://github.com/kcp-dev/kcp) repository
+- Staging repository: a repository that is part of the monorepo
+  (located within `kcp-dev/kcp/staging/src/github.com/kcp-dev`)
+- Published repository: a standalone repository that's a mirror of the staging repository
+
+Note: the term 'staging repository' does **not** imply a Git repository (with `.git` directory), but an
+authoritative place that stores code. Usually, each staging repository is a standalone Go module.
```

Comment:
`Usually` - Is this case for all our modules/libs? If so, maybe we should say - every stagging sub-repo is a separate go module.